sdk/identity/identity/src/credentials/azurePipelinesCredential.ts

```diff
@@ -120,34 +120,40 @@ export class AzurePipelinesCredential implements TokenCredential {
           response.status
         }. Complete response - ${JSON.stringify(response)}`,
       );
-      throw new AuthenticationError(
-        response.status,
-        `${credentialName}: Authenticated Failed. Received null token from OIDC request. Response status- ${
-          response.status
-        }. Complete response - ${JSON.stringify(response)}`,
-      );
+      throw new AuthenticationError(response.status, {
```

Comment:
Can we add tests for these? I think there's already a framework for mocking identityClient request/responses that we can possibly use.

If not, we can at least move some of this logic out to a separate function so that it is more testable.

Here's a test I would love to see:

- A response without text throws the expected error (including the status and error_description)
- A response with valid json but no oidcToken throws the expected error

This will make reviewing easier as well since it'll be easy to see what the result object looks like. 

I'd say timebox two hours to add tests and if you get stuck let me know and I will play around with it as well - would you be agreeable to that?

Reply:
done